app/controllers/users_controller.rb

```diff
@@ -47,6 +48,29 @@ def index
     end
   end
 
+  private def check_edit_access
+    @user = User.find(params.require(:id))
+    cannot_edit_reason = current_user.cannot_edit_data_reason_html(@user)
```

Comment:
Are we sure that the user is definitely always logged in at this point? Otherwise it should be `current_user&.foo(..)`

Reply:
Yes the user will be logged in because there is `before_action :authenticate_user!` before this `before_action`.